Python/ceval.c

```diff
@@ -5238,8 +5238,17 @@ import_from(PyThreadState *tstate, PyObject *v, PyObject *name)
         PyErr_SetImportError(errmsg, pkgname, NULL);
     }
     else {
+        _Py_IDENTIFIER(__spec__);
+        PyObject* spec = _PyObject_GetAttrId(v, &PyId___spec__);
+        Py_XINCREF(spec);
+        const char* fmt =
+            _PyModuleSpec_IsInitializing(spec) ?
+            "cannot import name %R from partially initialized module %R (most likely due to a circular import) (%S)" :
+            "cannot import name %R from %R (%S)";
+        Py_XDECREF(spec);
+
         errmsg = PyUnicode_FromFormat(
-            "cannot import name %R from %R (%S)",
+            fmt,
```

Comment:
Please reformat this function call such that all of the arguments are on one line.

Reply:
hmm I thought it was slightly more readable with the format variables and the format string separate, but sure